linkml_runtime/utils/dataclass_extensions_376.py

```diff
@@ -1,30 +1,54 @@
+import sys
 import dataclasses
-#
-# The dataclass library builds a rigid `__init__` function that doesn't allow any unrecognized named parameters
-#
-# The purpose of this extension is to enhance the library to allow additional keyword arguments to passed in
-# and then on to the __post_init__ function that can deal with them accordingly
+import warnings
 
-# Beware that there is no promise that signature of the create function will remain consistent
-loc_fn = dataclasses._create_fn
+# Block unsupported Python versions
+if sys.version_info >= (3, 13):
+    raise RuntimeError(
+        "The LinkML dataclass_extensions_376 patch is no longer compatible with Python 3.13 or newer.\n\n"
+        "Python 3.13 removed internal support for `_create_fn`, which this patch depends on.\n\n"
+        "To resolve this:\n"
+        "  • Upgrade your LinkML schema and code to use LinkML >= 1.9.0, which no longer uses this patch\n"
+        "  • Or migrate to Pydantic models for dataclass behavior\n\n"
+    )
 
+# Warn that this extension is deprecated
+warnings.warn(
+    "The LinkML dataclass extension patch is deprecated and will be removed in a future release.\n"
+    "Consider upgrading to LinkML >= 1.9.0 or switching Python class generation to Pydantic.",
```

Comment:
```suggestion
    "The LinkML dataclass extension patch is deprecated and will be removed in v1.10.0.\n"
    "To upgrade, either:\n"
    "- Regenerate this module with linkml >=1.9.0 (preferred)\n"
    "- Remove the import of dataclass_extensions_376 manually, or\n"
    "- Consider switching Python class generation to Pydantic, which will eventually replace python dataclasses.",
```

usually good to give a specific version we're going to remove it in, and just adding more detail on the upgrade path.

Reply:
No, I am not ready to set a specific deprecation release, yet.

Reply:
i guess i'm a little unclear on the plan here then - i assumed we would just be marking this as deprecated now and going with the standard dataclass impl, which is functionally equivalent, so i'm not really sure why not set a removal version, but up to you

Reply:
We need to establish a release process that incorporates checks for deprecation removals as part of each release. Additionally, we should create small, independent pull requests (PRs) to handle these removals. It would be beneficial to coordinate the timing of these removals as an extra bonus (for example, we should ask if folks have bandwidth, assign the removal tasks, and remind folks to get the deprecation PRs in when a release is coming - and further it should be predictable when a release is coming - we lost the plot on this particular bit for 1.9.0 and will need to bump the "removed_in" annotations for a few things).  <-- all of this is in progress at the moment.  

I appreciate the concept and the implementation you’ve developed, and the tests that support it.